packages/block-library/src/buttons/test/edit.native.js

```diff
@@ -232,6 +243,45 @@ describe( 'Buttons block', () => {
 
 				expect( getEditorHtml() ).toMatchSnapshot();
 			} );
+
+			it( 'removes the button and buttons block when deleting the block using the delete (backspace) key', async () => {
+				const screen = await initializeEditor( {
+					initialHtml: BUTTONS_HTML,
+				} );
+
+				// Get block
+				const buttonsBlock = await getBlock( screen, 'Buttons' );
+
+				// Trigger inner blocks layout
+				const innerBlockListWrapper = await within(
+					buttonsBlock
+				).findByTestId( 'block-list-wrapper' );
+				fireEvent( innerBlockListWrapper, 'layout', {
+					nativeEvent: {
+						layout: {
+							width: 300,
+						},
+					},
+				} );
+
+				// Get inner button block
+				const buttonBlock = await getBlock( screen, 'Button' );
+				fireEvent.press( buttonBlock );
+
+				// Delete block
+				fireEvent( buttonBlock, 'onKeyDown', {
+					nativeEvent: {},
+					preventDefault() {},
+					keyCode: BACKSPACE,
+				} );
+
+				expect( getEditorHtml() ).toMatchSnapshot();
+
+				const component = getTestComponentWithContent();
+
+				const rendered = component.toJSON();
+				expect( rendered ).toMatchSnapshot();
```

Comment:
@derekblank Not sure what's the purpose of testing here the `Button` component (note we are using here [the component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/button/index.native.js), not [the block](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/button/block.json)), could you elaborate on the purpose of covering this? Thanks 🙇 !

Reply:
The initial intend was to better capture that the app would actually crash, rather than just check the HTML. After @geriux clarified that we should move the prop check [here](https://github.com/WordPress/gutenberg/pull/51435#discussion_r1227768517), I've removed this `getTestComponentWithContent` check.